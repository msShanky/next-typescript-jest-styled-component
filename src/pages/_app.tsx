import type { AppProps } from 'next/app'
import { createGlobalStyle, ThemeProvider } from 'styled-components'
import normalize from 'styled-normalize'
import { mindBuilderTheme } from 'mind-builder.theme'
import '../styles/font.css'

const GlobalStyle = createGlobalStyle`
  ${normalize}
  body {
    font-family : ${({ theme }) => theme.fontFamily.regular}
  }
`

function MyApp({ Component, pageProps }: AppProps) {
  return (
    <ThemeProvider theme={mindBuilderTheme}>
      <GlobalStyle />
      <Component {...pageProps} />
    </ThemeProvider>
  )
}

export default MyApp
