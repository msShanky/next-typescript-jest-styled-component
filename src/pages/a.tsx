import { ReactElement } from 'react'

export default function A(): ReactElement {
  return <div>a</div>
}
