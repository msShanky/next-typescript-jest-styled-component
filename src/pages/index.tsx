import Head from 'next/head'
import styled from 'styled-components'

const Container = styled.div`
  background: white;
  display: grid;
  grid-template-columns: 20% 1fr 20%;
  grid-template-rows: 1fr;
`
// const ComponentPanel = styled.div``

export const Home = (): JSX.Element => (
  <Container>
    <Head>
      <title>Create Next App</title>
      <link rel="icon" href="/favicon.ico" />
    </Head>
    <div>
      <p>This is the left panel</p>
    </div>
    <div>
      <p>This is the canvas</p>
    </div>
    <div>
      <p>This is properties panel</p>
    </div>
  </Container>
)

export default Home
