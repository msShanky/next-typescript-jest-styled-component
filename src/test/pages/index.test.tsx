import { Home } from '../../pages/index'
import React from 'react'
import { render } from '../testUtilities'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('to have left panel', () => {
    const { getByText } = render(<Home />)
    expect(getByText('This is the left panel')).toBeInTheDocument()
  })
  it('to have canvas panel', () => {
    const { getByText } = render(<Home />)
    expect(getByText('This is the canvas')).toBeInTheDocument()
  })
  it('to have properties panel', () => {
    const { getByText } = render(<Home />)
    expect(getByText('This is properties panel')).toBeInTheDocument()
  })
})
