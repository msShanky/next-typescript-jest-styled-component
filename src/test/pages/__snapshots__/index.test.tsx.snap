// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pages__Container-sc-86lw4u-0 jRQwTN"
  >
    <div>
      <p>
        This is the left panel
      </p>
    </div>
    <div>
      <p>
        This is the canvas
      </p>
    </div>
    <div>
      <p>
        This is properties panel
      </p>
    </div>
  </div>
</DocumentFragment>
`;
